src/Strategy/NuGet/ProjectAssetsJson.hs

```diff
@@ -0,0 +1,93 @@
+module Strategy.NuGet.ProjectAssetsJson
+  ( discover
+  , strategy
+  , buildGraph
+  , analyze
+
+  , ProjectAssetsJson(..)
+  ) where
+
+import Prologue
+
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import           Polysemy
+import           Polysemy.Input
+import           Polysemy.Output
+
+import           DepTypes
+import           Discovery.Walk
+import           Effect.ReadFS
+import           Graphing (Graphing, unfold)
+import           Types
+
+discover :: Discover
+discover = Discover
+  { discoverName = "projectassetsjson"
+  , discoverFunc = discover'
+  }
+
+discover' :: Members '[Embed IO, Output ConfiguredStrategy] r => Path Abs Dir -> Sem r ()
+discover' = walk $ \_ subdirs files -> do
+  case find (\f -> fileName f == "project.assets.json") files of
+    Just file -> output (configure file)
+    Nothing -> pure ()
+
+  walkSkipNamed ["node_modules/"] subdirs
+
+strategy :: Strategy BasicFileOpts
+strategy = Strategy
+  { strategyName = "nuget-projectassetsjson"
+  , strategyAnalyze = \opts -> analyze
+      & fileInputJson @ProjectAssetsJson (targetFile opts)
+  , strategyModule = parent. targetFile
+  , strategyOptimal = NotOptimal
+  , strategyComplete = NotComplete
+  }
+
+data ProjectAssetsJson = ProjectAssetsJson
+  { targets     :: M.Map Text (M.Map Text NugetDep)
+  } deriving Show
+
+instance FromJSON ProjectAssetsJson where
+  parseJSON = withObject "ProjectAssetsJson" $ \obj ->
+    ProjectAssetsJson <$> obj .: "targets"
+
+data NugetDep = NugetDep
+  { depType    :: Text
+  , deepDeps   :: Maybe (M.Map Text Text)
```

Comment:
I'd recommend making this field just `deepDeps :: M.Map Text Text` -- it'll make this easier to consume in `buildGraph`

in the FromJSON parser below, you can use `(.!=) :: Parser (Maybe a) -> a -> Parser a` like:
```
instance FromJSON NugetDep where
  parseJSON = withObject "Dependency" $ \obj ->
    NugetDep <$> obj .: "type"
             <*> obj .:? "dependencies" .!= M.empty
```

Reply:
This is a really great solution and saved me from having to care about the Maybe as well. Good catch, thanks!